packages/webdriver/src/index.ts

```diff
@@ -152,11 +151,11 @@ export default class WebDriver {
      * @returns {string}           the new session id of the browser
      */
     static async reloadSession(instance: Client, newCapabilities?: WebdriverIO.Capabilities) {
-        const capabilities = deepmerge(instance.requestedCapabilities, newCapabilities || {})
-        const params: Options.WebDriver = { ...instance.options, capabilities }
+        const capabilities = { ...sanitizeCaps(newCapabilities || {}, true), capabilities: sanitizeCaps(newCapabilities || instance.requestedCapabilities) }
+        const params: Options.WebDriver = { ...instance.options, ...capabilities }
```

Comment:
This would merge capabilities into the `params` object which should be of type `Options.WebDriver`, can you explain your intention doing this? It feels like this is not what we want, right?

Reply:
Yeah, the merging is what I aimed for. Because of this error `Error: Invalid or unsupported WebDriver capabilities found ("hostname", "port").`, I need a way to separate out fields like host, port, etc.. from an object that is passed into reloadFunction and use them to overwrite the old values existed in `instance.options`. So, assuming you pass something like this in reloadFunction,
```
{
host: 'xxxx',
port: 1234,
browserName: 'chrome'
}
```
That top line would make `capabilities` field look like this
```
host: 'xxxx',
port: 1234,
capabilities: {
    browserName: 'chrome'
}
```

And I can just use this to overwrite the corresponding value in `instance.options`.

Not sure if I answer your question, but let me know.

Reply:
I would suggest to just have some simple if statements and move the connection parameters into the params object, e.g.:

```ts
for (const prop of ['protocol', 'hostname', 'port', 'path'] as Option.Connection) {
    if (prop in capabilities) {
        params[prop] = capabilities[prop];
        delete capabilities[prop];
    }
}
```

Reply:
better?